### ISSUE TITLE
Incomplete handling of VarArgs in dataflow

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#1218]
Consider a method that has a varargs parameter:

```java
void foo(String... args) {}
```

`javac` implicitly converts a call to `foo` with any number of string arguments into an array. For instance, `foo()` is actually `foo(new String[0])`, while `foo("bar", "baz")` is actually `foo(new String {"bar", "baz"})`. If `foo` is called with a literal array of strings, then that's what's passed.

When the CFG is constructed in `CFGBuilder`, the Checker Framework appropriately creates a new, artificial `ArrayCreationNode` to represent the new array, if one is implicitly created. However, the Checker Framework does **not** create a new AST to go along with it: instead, a `null` tree is inserted. Most of the time, this is fine. However, consider a situation where a programmer writes an annotation on the varargs parameter (borrowing the annotation from the Index Checker):

```java
void foo(String @MinLen(1) ... args) {
   // Do something with args[0]
}
```

The annotation says that the implicitly created array must have at least one element. If an actual array is passed when the function is called, it's checked correctly:
```java
String[] args = new String[0];
foo(args); // incompatible assignment, warning issued
```

But, if the function is called using varargs syntax, no warning is issued - because there is no tree to visit. The annotated type factory can't assign a type to a null tree, and the transfer function can't insert a type into the store for a node that has a null tree. So, for instance, the call `foo()` never has the type of its argument checked, and is accepted without a warning by the Checker Framework.

I think that the solution is to create an artificial tree in `CFGBuilder` using the `TreeBuilder` interface, in much the same way that artificial trees are created for enhanced for loops and other constructions that include implicit CFG nodes.

**Comment by wmdietl** [comment]
I'm not sure that the same approach as for enhanced for loops would work here: the SourceVisitor would still have no argument tree for `foo()` and would still not visit anything. Adding the AST to the created argument in the CFG won't be visible to the AST visitor.

Minor comment: the method signature should be 'void foo(String @MinLen(1) ... args) {' to express the minimum length of the array.

**Comment by kelloggm** [comment]
I've fixed the typo in the code example.
